src/metatensor/models/utils/neighbors_lists.py

```diff
@@ -0,0 +1,99 @@
+from typing import List, Union
+
+import torch
+from metatensor.learn.data.dataset import _BaseDataset
+from metatensor.torch import Labels, TensorBlock
+from metatensor.torch.atomistic import (
+    NeighborsListOptions,
+    System,
+    register_autograd_neighbors,
+)
+from rascaline.torch import NeighborList
+
+
+REQUIRED_NL_SAMPLES = [
+    "first_atom",
+    "second_atom",
+    "cell_shift_a",
+    "cell_shift_b",
+    "cell_shift_c",
+]
+
+
+def get_rascaline_neighbors_list(
+    systems: Union[System, List[System]], options: NeighborsListOptions
+) -> Union[TensorBlock, List[TensorBlock]]:
+    """
+    Calculates the neighborlist for a given system using
+    rascaline.NeghborList calculator. Additionally, it registers
+    the neighborlist for torch autograd.
+
+    :param systems: A single systems or a list systems.
+    :param options: A NeighborsListOptions object.
+
+    :return: A TensorBlock or a list of TensorBlocks containing neigborlists
+    information.
+    """
+
+    if not isinstance(systems, list):
+        systems = [systems]
+    nl_calculator = NeighborList(
+        cutoff=options.model_cutoff, full_neighbor_list=options.full_list
+    )
+    nl_list = []
+    for system in systems:
+        nl_tmap = nl_calculator.compute(system)
+        tmp_nl = nl_tmap.keys_to_samples(nl_tmap.keys.names).block()
+        required_indices = [
+            i
+            for i in range(len(tmp_nl.samples.names))
+            if tmp_nl.samples.names[i] in REQUIRED_NL_SAMPLES
+        ]
+        samples = Labels(
+            names=REQUIRED_NL_SAMPLES, values=tmp_nl.samples.values[:, required_indices]
+        )
```

Comment:
```suggestion
        samples = tmp_nl.samples.remove("structure")
```

Longer term, this might be doable as `mts.remove_dimension_block(nl, axis="samples", "structure")`; but we don't have this operation yet.